src/editors/sharedComponents/InsertLinkModal/index.jsx

```diff
@@ -0,0 +1,170 @@
+import { useEffect, useState } from 'react';
+import PropTypes from 'prop-types';
+import { logError } from '@edx/frontend-platform/logging';
+import { useIntl } from '@edx/frontend-platform/i18n';
+import {
+  Button,
+  Tabs,
+  Tab,
+  Form,
+} from '@openedx/paragon';
+import BaseModal from '../BaseModal';
+import BlocksList from './BlocksList';
+import BlockLink from './BlockLink';
+import SearchBlocks from './SearchBlocks';
+import { formatBlocks, isValidURL } from './utils';
+import { getBlocksFromCourse } from './api';
+
+import messages from './messages';
+import './index.scss';
+
+const InsertLinkModal = ({
+  courseId,
+  isOpen,
+  onClose,
+  editorRef,
+}) => {
+  const intl = useIntl();
+  const [searchField, setSearchField] = useState('');
+  const [blocksSearched, setBlocksSearched] = useState(false);
+  const [blockSelected, setBlocksSelected] = useState(null);
+  const [blocksList, setBlocksList] = useState(null);
+  const [invalidUrlInput, setInvalidUrlInput] = useState(false);
+  const [inputUrlValue, setInputUrlValue] = useState('');
+
+  const handleSearchedBlocks = (isSearched) => {
+    setBlocksSearched(isSearched);
+  };
+
+  const handleSelectedBlock = (blockSelectedFromList) => {
+    setBlocksSelected(blockSelectedFromList);
+    setInputUrlValue('');
+  };
+
+  const handleCloseLink = () => {
+    setSearchField('');
+    setBlocksSelected(null);
+  };
+
+  const handleChangeInputUrl = ({ target: { value } }) => {
+    setInputUrlValue(value);
+  };
+
+  /* istanbul ignore next */
+  const handleSave = () => {
+    const editor = editorRef.current;
+    const urlPath = blockSelected?.lmsWebUrl || inputUrlValue;
+    if (editor && urlPath) {
+      const validateUrl = isValidURL(urlPath);
+
+      if (!validateUrl) {
+        setInvalidUrlInput(true);
+        return;
+      }
+
+      const selectedText = editor.selection.getContent({ format: 'text' });
+
+      if (selectedText.trim() !== '') {
+        const linkHtml = `<a href="${urlPath}" data-mce-href="${urlPath}" target="_blank">${selectedText}</a>`;
+        editor.selection.setContent(linkHtml);
+      }
```

Comment:
### Product

@felipemontoya @santiagosuarezedunext - the current implementation always shows the "Insert link" button in the toolbar

![image](https://github.com/openedx/frontend-lib-content-components/assets/112954497/19dc1276-8cb2-4b9f-aaae-711d89c32e3a)

but this logic *requires* a user to have selected text in order to create a link. This results in a silent failure when trying to add a link without any selected text.

### Techinical

@johnvente I also found what feels like a bug here. If you create a new at first everything seems fine

![image](https://github.com/openedx/frontend-lib-content-components/assets/112954497/e7847eb6-083f-4cbc-9651-3cb3a05f832c)

![image](https://github.com/openedx/frontend-lib-content-components/assets/112954497/918eb5b4-3261-4dae-9f83-ed8be4e5e35d)

but then, if you select the link text and click "Insert link" again you're shown a new modal with no URL. Entering a URL and clicking save results in nested links

![image](https://github.com/openedx/frontend-lib-content-components/assets/112954497/e3bbb374-2f2a-4ae7-aecb-6ea2cd6353af)